src/_path.h

```diff
@@ -981,6 +981,56 @@ char *__append_to_string(char *p, char **buffer, size_t *buffersize,
     return p;
 }
 
+
+char *__add_number(double val, const char *format, int precision,
+                 char **buffer, char *p, size_t *buffersize)
+{
+    char *result;
+
+#if PY_VERSION_HEX >= 0x02070000
+    char *str;
+    str = PyOS_double_to_string(val, format[0], precision, 0, NULL);
+#else
+    char str[64];
+    PyOS_ascii_formatd(str, 64, format, val);
+#endif
+
+    // Delete trailing zeros and decimal point
+    char *q = str;
+    for (; *q != 0; ++q) {
```

Comment:
Yes.  PyOS_double_to_string and PyOS_ascii_formatd return NULL-terminated strings.  The docs don't actually say that, but since they don't return a length, I don't see how one could expect anything else.